addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -94,9 +89,9 @@ def _compute_state(self):
         batchs = self.filtered(lambda batch: batch.state not in ['cancel', 'done'])
         for batch in batchs:
             if not batch.picking_ids:
-                return
+                batch._origin.state = 'cancel'
             # Cancels automatically the batch picking if all its transfers are cancelled.
-            if all(picking.state == 'cancel' for picking in batch.picking_ids):
+            elif all(picking.state == 'cancel' for picking in batch.picking_ids):
```

Comment:
Why on `batch._origin.state` and not on `batch.state` ?
Can you do like this ?
```python
# Cancels the batch picking if it has no transfers or if they are all cancelled.
if not batch.picking_ids or all(picking.state == 'cancel' for picking in batch.picking_ids):
```
Also, in the commit message, instead of explain what the code does, it could be better to explain why.
It's only a suggestion, but maybe something like that:
> As pickings can be moved from a batch into an another one, a batch can suddenly be emptied or having no more active pickings. In such case, we cancel it.
> This way, empty batches won't be displayed by default as there is a default filter showing only draft or in progress batches.

Reply:
Last time when I was doing it, self was always "new id". Now I checked it's not. I must be drunk. 
I applied all the change.